virtio-devices/src/vhost_user/vu_common_ctrl.rs

```diff
@@ -108,13 +112,35 @@ pub fn setup_vhost_user<S: VhostUserMasterReqHandler>(
     virtio_interrupt: &Arc<dyn VirtioInterrupt>,
     acked_features: u64,
     slave_req_handler: &Option<MasterReqHandler<S>>,
+    inflight: Option<&mut Inflight>,
 ) -> Result<()> {
     vu.set_features(acked_features)
         .map_err(Error::VhostUserSetFeatures)?;
 
     // Let's first provide the memory table to the backend.
     update_mem_table(vu, mem)?;
 
+    // Setup for inflight I/O tracking shared memory.
+    if let Some(inflight) = inflight {
+        if inflight.fd.is_none() {
+            let inflight_req_info = VhostUserInflight {
+                mmap_size: 0,
+                mmap_offset: 0,
+                num_queues: queues.len() as u16,
+                queue_size: queues[0].actual_size(),
+            };
+            let (info, fd) = vu
+                .get_inflight_fd(&inflight_req_info)
+                .map_err(Error::VhostUserGetInflight)?;
+            inflight.info = info;
+            inflight.fd = Some(fd);
+        }
+        if let Some(inflight_fd) = inflight.fd.as_ref() {
+            vu.set_inflight_fd(&inflight.info, inflight_fd.as_raw_fd())
+                .map_err(Error::VhostUserSetInflight)?;
+        }
```

Comment:
Ok one more nit but I think we can simplify further to make this clear:
```suggestion
        // Unwrapping the inflight fd is safe here since we know it can't be None.
        vu.set_inflight_fd(&inflight.info, inflight.fd.as_ref.unwrap().as_raw_fd()).map_err(Error::VhostUserSetInflight)?;
```

Reply:
I think this makes clear the fact that SET will always be called, while GET is only called the first time.